EIPS/eip-1474.md

```diff
@@ -2147,7 +2147,7 @@ _(none)_
 
 ##### Returns
 
-{`boolean|object`} - `false` if this client is not syncing with the network, otherwise an object with the following members:
+{`boolean|object`} - `false` if `currentBlock` is equal to `highestBlock` (as defined below), otherwise an object with the following members:
```

Comment:
We should figure out what all of the clients do to determine the return value here.  Does someone want to volunteer to do this research?

I suspect it has more to do with whether syncing is forcibly stopped by the user, or if there is something preventing further syncing (e.g., no peers) than just being transiently behind.  That is, a client can be syncing:true when it starts syncing and is way behind, or when a new block has just arrived and it hasn't processed it yet.

Reply:
Apparently, Nethermind doesn't returns false even when `currentBlock` and `highestBlock` are equal. So clients implementations seem to differ from each other on this.
Another possibility to fix the ambiguity is to completely remove the `boolean` return type.

Reply:
So, here's what Nethermind is currently doing. The RPC method is implemented in [EthModule.cs#L77](https://github.com/NethermindEth/nethermind/blob/eed064c478f53994c8590b5d2203479389ed92c3/src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs#L77) -> 
```csharp
public ResultWrapper<SyncingResult> eth_syncing()
{
    SyncingResult result;
    if (_blockchainBridge.IsSyncing)
    {
        result = new SyncingResult
        {
            CurrentBlock = _blockchainBridge.Head.Number,
            HighestBlock = _blockchainBridge.BestKnown,
            StartingBlock = 0L,
            IsSyncing = true
        };
    }
    else
    {
        result = SyncingResult.NotSyncing;
    }

    return ResultWrapper<SyncingResult>.Success(result);
}
```
And here's the `IsSyncing` property definition in [BlockchainBridge.cs#L124](https://github.com/NethermindEth/nethermind/blob/eed064c478f53994c8590b5d2203479389ed92c3/src/Nethermind/Nethermind.Facade/BlockchainBridge.cs#L124) -> 
```csharp
public bool IsSyncing => _blockTree.BestSuggestedHeader.Hash != _blockTree.Head.Hash;
```
The actual calculations for `BestSuggestedHeader` and `Head` are somewhat complicated and implemented in [BlockTree.cs](https://github.com/NethermindEth/nethermind/blob/eed064c478f53994c8590b5d2203479389ed92c3/src/Nethermind/Nethermind.Blockchain/BlockTree.cs). I'll continue to post my findings here.